src/routes/activities/+page.svelte

```diff
@@ -1,7 +1,34 @@
 <script lang="ts">
-	import { page } from '$app/stores';
+	import { ActivityCard } from '$lib/components/cards';
+	import { searchHandler } from '$lib/utils';
+	import { _ } from 'svelte-i18n';
+	import type { PageData } from './$types';
+
+	export let data: PageData;
+
+	const searchActivities = data.activities.map((activity) => ({
+		...activity,
+		searchTerms:
+			`${activity.title} ${activity.description} ${activity.faculty} ${activity.location} ${activity.prerequisites}`.replaceAll(
+				'null',
+				''
+			)
+	}));
+
+	$: filteredResults = searchHandler(data.query, searchActivities);
 </script>
 
 <div>
-	{$page.url.searchParams.get('q')}
+	<h1 class="text-4xl text-secondary font-semibold text-center mb-12">{$_('a-default.title')}</h1>
+	{#if filteredResults.length > 0}
+		<div class="grid md:grid-cols-2 xl:grid-cols-3 gap-10 justify-center">
+			{#each filteredResults as activity}
+				<ActivityCard activity={{ ...activity }} />
```

Comment:
I believe in Svelte we can do this:
```suggestion
				<ActivityCard {activity} />
```